django/core/management/sql.py

```diff
@@ -2,11 +2,14 @@
 from django.db import models
 
 
-def sql_flush(style, connection, reset_sequences=True, allow_cascade=False):
+def sql_flush(style, connection, reset_sequences=True, allow_cascade=False,
+              only_nonempty_tables=False):
     """
     Return a list of the SQL statements used to flush the database.
     """
     tables = connection.introspection.django_table_names(only_existing=True, include_views=False)
+    if only_nonempty_tables:
+        tables = connection.introspection.only_nonempty_tables(tables)
```

Comment:
```suggestion
    tables = connection.introspection.django_table_names(only_existing=True, include_views=False)
    if skip_empty:
        tables = connection.introspection.exclude_empty_tables(tables)
```

Alternatively, because this is quite specific, I think that we could just make this a flag on `.django_table_names()`:

```suggestion
    tables = connection.introspection.django_table_names(
        only_existing=True,
        include_views=False,
        skip_empty=skip_empty,
    )
```

But I'm happy either way if others think not.